entity.go

```diff
@@ -36,6 +36,9 @@ type Entity interface {
 	// desc and ext. The ok value will be true if the extension was found.
 	Extension(desc *proto.ExtensionDesc, ext interface{}) (ok bool, err error)
 
+	// Extensions returns all of the Extensions on this entity.
+	Extensions() []Extension
```

Comment:
This should only be on `Message`, not `Entity` as only Messages can be extended. Also make it clear that these extensions aren't necessarily defined within the message, but rather applied to it.